### PR TITLE
Widget templating

### PR DESCRIPTION
**Pull request by Armanul46**

### CODE REVIEW

File: templates/widgets/all-locations.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * @author  wpWax
+ * @since   7.2.0
+ * @version 7.2.0
+ */
+
+use Directorist\Directorist_Listings;
```
Comment (kowsar89) [inline templates/widgets/all-locations.php:8]: remove this line from all files if unused

File: includes/widgets/all-categories.php [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+use Directorist\Helper;
+
+if ( ! defined( 'ABSPATH' ) ) exit;
+
+class All_Categories extends \WP_Widget {
+
+	public function __construct() {
+		$id_base        = 'bdcw_widget';
+        $name           = esc_html__( 'Directorist - Categories', 'directorist', 'directorist' );
```
Comment (kowsar89) [inline includes/widgets/all-categories.php:16]: double textdomain, this issue exists in other files as well

File: includes/widgets/all-tags.php [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        } else {
+            $tags = $this->atbdp_tags_list( $query_args );
+        }
+
+		Helper::get_template( 'widgets/all-tags', compact( 'args', 'instance', 'query_args', 'tags' ) );
+
+		echo wp_kses_post( $args['after_widget'] );
+	}
+
+    public function atbdp_tags_list( $settings ) {
```
Comment (kowsar89) [inline includes/widgets/all-tags.php:161]: use 'directorist' prefix instead of 'atbdp' in all methods, check other files too

**Review by kowsar89 — CHANGES_REQUESTED** [review]

File: directorist-base.php [hunk 4]
```diff
@@ -519,20 +523,20 @@ public function register_widgets()
...
-		register_widget('BD_Categories_Widget');
-		register_widget('BD_Locations_Widget');
-		register_widget('BD_Tags_Widget');
-		register_widget('BD_Search_Widget');
-		register_widget('BD_Map_Widget');
-		// register_widget('BD_All_Map_Widget');
-		register_widget('BD_Similar_Listings_Widget');
-		register_widget('BD_Author_Info_Widget');
-		register_widget('BD_Featured_Listings_Widget');
+		// register_widget('BD_Popular_Listing_Widget');
```
Comment (kowsar89) [inline directorist-base.php:526]: remove these comments

File: includes/model/Listings.php [hunk 5]
```diff
@@ -2014,4 +2014,111 @@ public static function mark_as_favourite_button() {
     		echo atbdp_listings_mark_as_favourite( get_the_ID() );
     	}
     }
+
+	public static function popular_listings_query( $count = 5 ) {
```
Comment (kowsar89) [inline includes/model/Listings.php:2018]: move this method inside widgets class

**Review by kowsar89 — CHANGES_REQUESTED** [review]

**Comment by kowsar89** [comment]
Also fix merge conflicts

**Review by kowsar89 — APPROVED** [review]
Code looks good, need front-end check

**Comment by vairafiq** [comment]
Screen Shot 2022-03-14 at 9 20 14 AM

Looks like version as second parameter in docs https://developer.wordpress.org/reference/functions/_deprecated_function/

File: directorist-base.php [hunk 6]
```diff
@@ -613,235 +602,12 @@ public function add_uninstall_menu($menus) {
...
-		}
-
-		if ( count( $meta_queries ) ) {
-			$meta_queries['relation'] = 'AND';
-			$args['meta_query'] = $meta_queries;
-		}
-
-		return new WP_Query( apply_filters( 'atbdp_popular_listing_args', $args ) );
+	public function show_popular_listing() {
+		_deprecated_function( '7.2.2', 'ATBDP()->show_popular_listing' );
```
Comment (kowsar89) [inline directorist-base.php:606]: wrong usage of _deprecated_function, https://developer.wordpress.org/reference/functions/_deprecated_function/

**Review by kowsar89 — CHANGES_REQUESTED** [review]

**Comment by vairafiq** [comment]
widget style breaks in theme's default sidebar

Alpha
Screen Shot 2022-03-16 at 11 13 40 AM

Current
Screen Shot 2022-03-16 at 11 14 52 AM

**Comment by vairafiq** [comment]
Screen Shot 2022-03-16 at 1 11 12 PM
Error in widget screen

**Pull request merged**